src/Microsoft.Health.Dicom.SqlServer/Features/Store/SqlIndexDataStore.cs

```diff
@@ -5,284 +5,99 @@
 
 using System;
 using System.Collections.Generic;
-using System.Data;
 using System.Threading;
 using System.Threading.Tasks;
 using Dicom;
 using EnsureThat;
-using Microsoft.Data.SqlClient;
-using Microsoft.Health.Dicom.Core.Exceptions;
-using Microsoft.Health.Dicom.Core.Extensions;
+using Microsoft.Health.Dicom.Core.Features.CustomTag;
 using Microsoft.Health.Dicom.Core.Features.Model;
 using Microsoft.Health.Dicom.Core.Features.Store;
 using Microsoft.Health.Dicom.Core.Models;
-using Microsoft.Health.Dicom.SqlServer.Features.Schema.Model;
 using Microsoft.Health.Dicom.SqlServer.Features.Storage;
 using Microsoft.Health.SqlServer.Features.Client;
-using Microsoft.Health.SqlServer.Features.Storage;
+using Microsoft.Health.SqlServer.Features.Schema;
 
 namespace Microsoft.Health.Dicom.SqlServer.Features.Store
 {
     internal class SqlIndexDataStore : IIndexDataStore
     {
         private readonly SqlIndexSchema _sqlServerIndexSchema;
         private readonly SqlConnectionWrapperFactory _sqlConnectionFactoryWrapper;
+        private readonly SchemaInformation _schemaInformation;
+        private IIndexDataStore _innerDataStore;
 
         public SqlIndexDataStore(
             SqlIndexSchema indexSchema,
-            SqlConnectionWrapperFactory sqlConnectionWrapperFactory)
+            SqlConnectionWrapperFactory sqlConnectionWrapperFactory,
+            SchemaInformation schemaInformation)
```

Comment:
The only concern I have is here is how little value this class actually provides- it's mostly a pass-through and seems to be more of an unnecessary initialization. The method Smitha had mentioned minimizes that overhead!

Reply:
I didn't find a good way to fit the static method with direct injection.

Theoretically, the static method should be called inside SqlServer layer instead of Core layer, since Schema version is something Core should not know.

The static method is like this:
`  
public static SqlIndexDataStore GetIndexDataStore(SchemaVersion version, SqlConnectionWrapperFactory sqlConnectionWrapperFactory)
        {
            switch (version)
            {
                case SchemaVersion.V1:
                    return new SqlIndexDataStoreV1(sqlConnectionWrapperFactory);
                case SchemaVersion.V2:
                case SchemaVersion.Unknown:
                default:
                    return new SqlIndexDataStoreV2(sqlConnectionWrapperFactory);
            }
        }
`

This method should be called in DicomServerBuilderSqlServerRegistrationExtensions.AddSqlServer, but how to pass SchemaVersion  and SqlConnectionWrapperFactory is something I didn't figure out.